contracts/upgradeable_contracts/OverdrawManagement.sol

```diff
@@ -4,28 +4,38 @@ import "../upgradeability/EternalStorage.sol";
 import "openzeppelin-solidity/contracts/math/SafeMath.sol";
 import "./Upgradeable.sol";
 import "./RewardableBridge.sol";
+import "./BasicBridge.sol";
 
-contract OverdrawManagement is EternalStorage, RewardableBridge, Upgradeable {
+contract OverdrawManagement is EternalStorage, RewardableBridge, Upgradeable, BasicBridge {
     using SafeMath for uint256;
 
     event UserRequestForSignature(address recipient, uint256 value);
 
-    function fixAssetsAboveLimits(bytes32 txHash, bool unlockOnForeign) external onlyIfUpgradeabilityOwner {
+    function fixAssetsAboveLimits(bytes32 txHash, bool unlockOnForeign, uint256 valueToUnlock)
+        external
+        onlyIfUpgradeabilityOwner
+    {
         require(!fixedAssets(txHash));
+        require(valueToUnlock <= maxPerTx());
         address recipient;
         uint256 value;
         (recipient, value) = txAboveLimits(txHash);
-        require(recipient != address(0) && value > 0);
-        setOutOfLimitAmount(outOfLimitAmount().sub(value));
+        require(recipient != address(0) && value > 0 && value >= valueToUnlock);
+        setOutOfLimitAmount(outOfLimitAmount().sub(valueToUnlock));
+        uint256 pendingValue = value.sub(valueToUnlock);
+        setTxAboveLimitsValue(pendingValue, txHash);
         if (unlockOnForeign) {
             address feeManager = feeManagerContract();
+            uint256 eventValue = valueToUnlock;
             if (feeManager != address(0)) {
-                uint256 fee = calculateFee(value, false, feeManager, HOME_FEE);
-                value = value.sub(fee);
+                uint256 fee = calculateFee(valueToUnlock, false, feeManager, HOME_FEE);
+                eventValue = valueToUnlock.sub(fee);
             }
-            emit UserRequestForSignature(recipient, value);
+            emit UserRequestForSignature(recipient, eventValue);
+        }
+        if (pendingValue == 0) {
```

Comment:
It make sense to perform this operation before the if-block where the fee manager is used. It could prevent possible re-entrance attack, if the code of the Fee Manager is compromised

Reply:
Good catch! Fixed here: https://github.com/poanetwork/poa-bridge-contracts/pull/247/commits/c38f023182894b1bcb6b10f4a363beabeba39e27

Reply:
thanks